source/1_engine/rex_engine/src/filesystem/path.cpp

```diff
@@ -0,0 +1,576 @@
+    {
+      rsl::vector<rsl::string_view> splitted_path = rsl::split(path, "/\\");
+      rsl::vector<rsl::string_view> splitted_start = rsl::split(start, "/\\");
+
+      auto res = rsl::mismatch(splitted_path.cbegin(), splitted_path.cend(), splitted_start.cbegin(), splitted_start.cend());
+      return rsl::join(res.lhs_it, splitted_path.cend(), rsl::string_view(&g_seperation_char)).as_string();
+    }
+    // Returns the latest access time of the file or directory at the given path
+    card64 get_access_time(rsl::string_view path)
+    {
+      rsl::win::handle file = internal::open_file_for_attribs(path);
+
+      // When we have an invalid handle, we return 0
+      if (!file.is_valid())
+      {
+        return 0;
+      }
+
+      FILETIME access_time{};
+      GetFileTime(file.get(), NULL, &access_time, NULL);
+      return rsl::win::to_integer(access_time);
+    }
+    // Returns the modification time of the file or directory at the given path
+    card64 get_modification_time(rsl::string_view path)
+    {
+      rsl::win::handle file = internal::open_file_for_attribs(path);
+
+      // When we have an invalid handle, we return 0
+      if (!file.is_valid())
+      {
+        return 0;
+      }
+
+      FILETIME modification_time{};
+      GetFileTime(file.get(), NULL, NULL, &modification_time);
+      return rsl::win::to_integer(modification_time);
+    }
+    // Returns the creation time of the file or directory at the given path
+    card64 get_creation_time(rsl::string_view path)
+    {
+      rsl::win::handle file = internal::open_file_for_attribs(path);
+
+      // When we have an invalid handle, we return 0
+      if (!file.is_valid())
+      {
+        return 0;
+      }
+
+      FILETIME creation_time{};
+      GetFileTime(file.get(), &creation_time, NULL, NULL);
+      return rsl::win::to_integer(creation_time);
+    }
+    // Returns the creation time of the file or directory at the given path
+    card64 get_file_size(rsl::string_view path)
+    {
+      rsl::win::handle file = internal::open_file_for_attribs(path);
+
+      // When we have an invalid handle, we simply return the input
+      if (!file.is_valid())
+      {
+        return -1;
+      }
+
+      DWORD high_word = 0;
+      DWORD low_word = GetFileSize(file.get(), &high_word);
+      return rex::merge_int32_to_int64(high_word, low_word);
+    }
+    // Returns if the given path has an extension
+    bool has_extension(rsl::string_view path)
+    {
+      return internal::extension_start(path) != rsl::string_view::npos();
+    }
+    // Returns if the path exists
+    bool exists(rsl::string_view path)
+    {
+      DWORD attribs = GetFileAttributesA(path.data());
+      return attribs != INVALID_FILE_ATTRIBUTES;
+    }
+    // Returns if the given path is an absolute path
+    bool is_absolute(rsl::string_view path)
+    {
+      if (path.length() < 4)
+      {
+        return false;
+      }
+
+      if (!rsl::is_alpha(path[0]))
```

Comment:
INFO: Is alpha?

Reply:
https://en.cppreference.com/w/cpp/string/byte/isalpha#:~:text=int%20isalpha(%20int%20ch%20)%3B,the%20currently%20installed%20C%20locale.